### PR TITLE
tools: add data synchronization intro

### PR DESCRIPTION
**Pull request by lilin90**
- Add the introduction to data synchronization
- Update the component name capitalization
- Update compatibility wording

@GregoryIan @csuzhangxc @CaitinChen PTAL
Via: https://github.com/pingcap/tidb-tools/blob/docs/docs/dm/zh_CN/user-manual.md

### CODE REVIEW

File: tools/data-migration-overview.md [hunk 1]
```diff
@@ -41,3 +41,29 @@ dmctl is the command line tool used to control the DM cluster.
...
+
+### Black and white lists synchronization at the schema and table levels
+
+The black and white lists filtering rule of the upstream database instances is similar to MySQL replication-rules-db/tables, which can be used to filter or only synchronize all operations of some databases or some tables.
+
+### Binlog event filtering
+
+Binlog event filtering is a more fine-grained filtering rule than the black and white lists synchronization at the schema and table levels. You can specify the database(s) or table(s) that you need to synchronize or filter out using statements like `INSERT`, `TRUNCATE TABLE`.
+
+### Column mapping filtering
```
Comment (csuzhangxc) [inline tools/data-migration-overview.md]:
```suggestion
### Column mapping
```

File: tools/data-migration-overview.md [hunk 2]
```diff
@@ -41,3 +41,29 @@ dmctl is the command line tool used to control the DM cluster.
...
+
+This section describes the data synchronization feature provided by Data Migration in detail.
+
+### Black and white lists synchronization at the schema and table levels
+
+The black and white lists filtering rule of the upstream database instances is similar to MySQL replication-rules-db/tables, which can be used to filter or only synchronize all operations of some databases or some tables.
+
+### Binlog event filtering
+
+Binlog event filtering is a more fine-grained filtering rule than the black and white lists synchronization at the schema and table levels. You can specify the database(s) or table(s) that you need to synchronize or filter out using statements like `INSERT`, `TRUNCATE TABLE`.
```
Comment (GregoryIan) [inline tools/data-migration-overview.md]: `the black and white lists synchronization`?  should it be `the black and white lists filtering rule`?

Comment (GregoryIan) [reply tools/data-migration-overview.md]: ` You can specify Binlog events of the database(s) or table(s) that you need to synchronize or filter...`

File: tools/data-migration-overview.md [hunk 3]
```diff
@@ -41,3 +41,29 @@ dmctl is the command line tool used to control the DM cluster.
...
+
+### Column mapping filtering
+
+Column mapping filtering is used to resolve the conflicts occurred when the sharding auto-increment primary key IDs are merged. The value of the auto-increment primary key ID can be modified according to the instance-id, which is configured by the user, and the schema/table name number.
```
Comment (GregoryIan) [inline tools/data-migration-overview.md]: `auto-increment primary key IDs are merged for shard tables`

Comment (GregoryIan) [reply tools/data-migration-overview.md]: and the schema/table ID

**Comment by GregoryIan** [comment]
LGTM

File: tools/data-migration-overview.md [hunk 4]
```diff
@@ -41,3 +41,29 @@ dmctl is the command line tool used to control the DM cluster.
...
+
+This section describes the data synchronization feature provided by Data Migration in detail.
+
+### Black and white lists synchronization at the schema and table levels
+
+The black and white lists filtering rule of the upstream database instances is similar to MySQL replication-rules-db/tables, which can be used to filter or only synchronize all operations of some databases or some tables.
+
+### Binlog event filtering
+
+Binlog event filtering is a more fine-grained filtering rule than the black and white lists filtering rule at the schema and table levels. You can specify the Binlog events of the database(s) or table(s) that you need to synchronize or filter out using statements like `INSERT`, `TRUNCATE TABLE`.
```
Comment (amyangfei) [inline tools/data-migration-overview.md]: `using statements like ...` is a little ambiguous, is it `You can specify ... using statements like ...` or `Binlog events using statements like...`

**Comment by lilin90** [comment]
@amyangfei PTAL again. Thanks!

**Pull request merged**